File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> lastInd (256, -1);
        int start = 0, maxLen = 0;
        for(int i=0;i<s.size();i++){
            if(lastInd[s[i]]>=start){
                start = lastInd[s[i]] + 1;
            }
            lastInd[s[i]] = i;
            maxLen = max(maxLen, i-start+1);
        } 
        return maxLen;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
    seen := map[byte]int{}
    left:=0
    longest:=0
    for right:=0; right<len(s);right++{
        c := s[right]
        if lastInd, ok := seen[c]; ok && lastInd>=left{
            left = max(left, lastInd+1)
        }
        seen[c] = right
        longest = max(longest, right-left+1)
    }
    return longest
}